### ISSUE TITLE
Issue with trainer.py deprecated

### ISSUE THREAD

**Issue by J03yP4nD4** [mindverse/Second-Me#177]
I've tried training a couple of times now and I keep getting stuck. I tried running make docker-restart-all and still came up with the same issue.
I'm at 80% and at the 4th step "Training to create Second Me"
Here is the log:

**** Running training *****
Num examples = 49
Num Epochs = 3
Instantaneous batch size per device = 2
Total train batch size (w. parallel, distributed & accumulation) = 2
Gradient Accumulation steps = 1
Total optimization steps = 75
Number of trainable parameters = 9,232,384
2025-04-09 06:02:01 [INFO] train.py:276 - === Training Begin ===
2025-04-09 06:02:01 [INFO] train.py:277 - Checking initial conditions:
0%| | 0/75 [00:00<?, ?it/s]2025-04-09 06:02:01 [INFO] train.py:276 - === Training Begin ===
2025-04-09 06:02:01 [INFO] train.py:277 - Checking initial conditions:
2025-04-09 06:02:01 [INFO] train.py:298 -
=== Starting Step 1 ===
2025-04-09 06:02:01 [INFO] train.py:298 -
2025-04-09 06:02:01 [INFO] train.py:304 - CPU Usage: 0.0%
=== Starting Step 1 ===
2025-04-09 06:02:01 [INFO] train.py:305 - Memory Usage: 759.50MB
2025-04-09 06:02:01 [INFO] train.py:304 - CPU Usage: 0.0%
2025-04-09 06:02:01 [INFO] train.py:308 - Thread Count: 62
2025-04-09 06:02:01 [INFO] train.py:305 - Memory Usage: 759.50MB
2025-04-09 06:02:01 [INFO] train.py:308 - Thread Count: 62
/usr/local/lib/python3.12/site-packages/transformers/trainer.py:3618: FutureWarning: `torch.cpu.amp.autocast(args...)` is deprecated. Please use `torch.amp.autocast('cpu', args...)` instead.
ctx_manager = torch.cpu.amp.autocast(cache_enabled=cache_enabled, dtype=self.amp_dtype)
2025-04-09 06:02:01 [INFO] trainprocess_service.py:832 - Training started
2025-04-09 06:02:01 [INFO] trainprocess_service.py:878 - Progress updated: 0% - Current step: 0/75
2025-04-09 06:02:59 [INFO] trainprocess_service.py:1236 - Training process has been requested to stop
2025-04-09 06:02:59 [INFO] trainprocess_service.py:1253 - Attempting to terminate process with PID: 27
2025-04-09 06:02:59 [INFO] trainprocess_service.py:1265 - Terminating child process with PID: 80
2025-04-09 06:02:59 [INFO] trainprocess_service.py:1265 - Terminating child process with PID: 81
2025-04-09 06:02:59 [INFO] script_runner.py:169 - Process (PID: 80) has ended, exit code: 0
Process (PID: 80) has ended, exit code: 0
2025-04-09 06:02:59 [INFO] trainprocess_service.py:793 - Training process started: {'pid': 80, 'environment': {'type': 'docker', 'details': 'docker-env-variable'}, 'log_file': '/app/logs/train/train.log', 'exit_code': 0}
2025-04-09 06:02:59 [INFO] trainprocess_service.py:1283 - All child processes of 27 have been terminated

**Comment by yingapple** [comment]
How much memory do you have?

**Comment by J03yP4nD4** [comment]
I'm using Windows and had to create a .wslconfig to change the allotted
memory. Docker was at less than 8GB and I had chosen the 16GB+
For this attempt even though I allocated 16GB, I decided to choose the 8GB+
instead for this attempt.
So while docker does show 15.16GB for available container memory.
I still am running into an error.
Thanks for reply and the help. It's much appreciated!

Here is the log:
**** Running training *****
Num examples = 52
Num Epochs = 3
Instantaneous batch size per device = 2
Total train batch size (w. parallel, distributed & accumulation) = 2
Gradient Accumulation steps = 1
Total optimization steps = 78
Number of trainable parameters = 4,399,104
2025-04-09 07:19:27 [INFO] train.py:276 - === Training Begin ===
2025-04-09 07:19:27 [INFO] train.py:277 - Checking initial conditions:
0%| | 0/78 [00:00<?, ?it/s]2025-04-09 07:19:27 [INFO] train.py:276 - ===
Training Begin ===
2025-04-09 07:19:27 [INFO] train.py:277 - Checking initial conditions:
2025-04-09 07:19:27 [INFO] train.py:298 -
=== Starting Step 1 ===
2025-04-09 07:19:27 [INFO] train.py:298 -
2025-04-09 07:19:27 [INFO] train.py:304 - CPU Usage: 0.0%
=== Starting Step 1 ===
2025-04-09 07:19:27 [INFO] train.py:305 - Memory Usage: 806.09MB
2025-04-09 07:19:27 [INFO] train.py:304 - CPU Usage: 0.0%
2025-04-09 07:19:27 [INFO] train.py:308 - Thread Count: 22
2025-04-09 07:19:27 [INFO] train.py:305 - Memory Usage: 806.09MB
2025-04-09 07:19:27 [INFO] train.py:308 - Thread Count: 22
/usr/local/lib/python3.12/site-packages/transformers/trainer.py:3618:
FutureWarning: `torch.cpu.amp.autocast(args...)` is deprecated. Please use
`torch.amp.autocast('cpu', args...)` instead.
ctx_manager = torch.cpu.amp.autocast(cache_enabled=cache_enabled,
dtype=self.amp_dtype)
2025-04-09 07:19:27 [INFO] trainprocess_service.py:832 - Training started
2025-04-09 07:19:27 [INFO] trainprocess_service.py:878 - Progress updated:
0% - Current step: 0/78
2025-04-09 07:20:00 [INFO] trainprocess_service.py:1019 - Weight merge task
result: {'returncode': 1, 'error': 'Execution failed, return code: 1'}
2025-04-09 07:20:00 [ERROR] trainprocess_service.py:1024 - Merge weights
failed: Execution failed, return code: 1
2025-04-09 07:20:00 [ERROR] trainprocess_service.py:1174 - Step
merge_weights failed
2025-04-09 07:20:00 [INFO] trainprocess_service.py:1175 - Marking step as
failed: stage=merge_weights, step=merge_weights

**Comment by yingapple** [comment]
@yexiangle Please check this issue, wrong training process management. So it skipped the training, and went into weight merging.

**Comment by J03yP4nD4** [comment]
I deleted the SecondMe I was working on, and this time only selected the 8GB Model. I used the same info as before, I only have 3 text notes. The longest note is 3 sentences.
It was looking promising the training second me got around 30% before falling back down to 0%. Having the weight merging problem again.

***** Running training *****
Num examples = 49
Num Epochs = 3
Instantaneous batch size per device = 2
Total train batch size (w. parallel, distributed & accumulation) = 2
Gradient Accumulation steps = 1
Total optimization steps = 75
Number of trainable parameters = 4,399,104
2025-04-14 01:32:54 [INFO] train.py:276 - === Training Begin ===
0%| | 0/75 [00:00<?, ?it/s]2025-04-14 01:32:54 [INFO] train.py:276 - === Training Begin ===
2025-04-14 01:32:54 [INFO] train.py:277 - Checking initial conditions:
2025-04-14 01:32:54 [INFO] train.py:277 - Checking initial conditions:
2025-04-14 01:32:54 [INFO] trainprocess_service.py:832 - Training started
2025-04-14 01:32:54 [INFO] trainprocess_service.py:878 - Progress updated: 0% - Current step: 0/75
2025-04-14 01:32:54 [INFO] train.py:298 -
=== Starting Step 1 ===
2025-04-14 01:32:54 [INFO] train.py:298 -
=== Starting Step 1 ===
2025-04-14 01:32:54 [INFO] train.py:304 - CPU Usage: 0.0%
2025-04-14 01:32:54 [INFO] train.py:304 - CPU Usage: 0.0%
2025-04-14 01:32:54 [INFO] train.py:305 - Memory Usage: 804.12MB
2025-04-14 01:32:54 [INFO] train.py:305 - Memory Usage: 804.12MB
2025-04-14 01:32:54 [INFO] train.py:308 - Thread Count: 22
2025-04-14 01:32:54 [INFO] train.py:308 - Thread Count: 22
/usr/local/lib/python3.12/site-packages/transformers/trainer.py:3618: FutureWarning: `torch.cpu.amp.autocast(args...)` is deprecated. Please use `torch.amp.autocast('cpu', args...)` instead.
ctx_manager = torch.cpu.amp.autocast(cache_enabled=cache_enabled, dtype=self.amp_dtype)
2025-04-14 01:32:57 [INFO] trainprocess_service.py:1019 - Weight merge task result: {'returncode': 1, 'error': 'Execution failed, return code: 1'}
2025-04-14 01:32:57 [INFO] trainprocess_service.py:1019 - Weight merge task result: {'returncode': 1, 'error': 'Execution failed, return code: 1'}
2025-04-14 01:32:57 [ERROR] trainprocess_service.py:1024 - Merge weights failed: Execution failed, return code: 1
2025-04-14 01:32:57 [ERROR] trainprocess_service.py:1024 - Merge weights failed: Execution failed, return code: 1
2025-04-14 01:32:57 [ERROR] trainprocess_service.py:1174 - Step merge_weights failed
2025-04-14 01:32:57 [ERROR] trainprocess_service.py:1174 - Step merge_weights failed
2025-04-14 01:32:57 [INFO] trainprocess_service.py:1175 - Marking step as failed: stage=merge_weights, step=merge_weights
2025-04-14 01:32:57 [INFO] trainprocess_service.py:1175 - Marking step as failed: stage=merge_weights, step=merge_weights
